src/FluentValidation/TestHelper/ValidatorTestExtensions.cs

```diff
@@ -48,6 +48,19 @@ public static class ValidationTestExtension {
 			return testValidationResult.ShouldHaveValidationErrorFor(expression);
 		}
 
+		public static IEnumerable<ValidationFailure> ShouldHaveValidationErrorFor<T>(this IValidator<T> validator, Expression<Func<T, T>> expression, string ruleSet)
+    where T : class, new()
+        {
+            T instance = Activator.CreateInstance<T>();
+            return validator.TestValidate<T>(instance, ruleSet).ShouldHaveAnyValidationError<T>();
+        }
+
+		public static IEnumerable<ValidationFailure> ShouldHaveValidationErrorFor<T>(this IValidator<T> validator, Expression<Func<T, T>> expression, T value, string ruleSet)
+		where T : class, new()
+		{
+			return validator.TestValidate<T>(value, ruleSet).ShouldHaveAnyValidationError<T>();

```

Comment:
`ShouldHaveAnyValidationError` is the wrong thing to be calling here. You need to be asserting that there's at least one validation failure at the model-level (ie that the property name is an empty string), which is done by calling `ShouldHaveValidationErrorFor(string.Empty);`